### PR TITLE
Improve crash analytics

### PR DESCRIPTION
**Pull request by summeroff**
required osn PR https://github.com/stream-labs/obs-studio-node/pull/687

### CODE REVIEW

File: app/services/crash-reporter.ts [hunk 1]
```diff
@@ -27,6 +27,7 @@ enum EAppState {
 
 interface ICrashReporterState {
   code: EAppState;
+  detected: string; // what was detected before close happens: ipc freez, backend crash, electron window unresponcive
```
Comment (EddyGharbi) [inline app/services/crash-reporter.ts:30]: I would rephrase the comment to: `The Backend event that was detected when the app close event was triggered: IPC freeze or backend crash or electron window unresponsive.`

File: app/services/crash-reporter.ts [hunk 2]
```diff
@@ -94,7 +95,14 @@ export class CrashReporterService extends Service {
...
 
   private writeStateFile(code: EAppState) {
-    this.appState = { code, version: this.version };
+    this.appState = { code, version: this.version, detected: '' };
+    this.appState = this.readStateFile();
+    this.appState.code = code;
+    this.appState.version = this.version;
+    if (this.appState.code !== EAppState.Starting) {
+      this.appState.detected = '';
+    }
```
Comment (holiber) [inline app/services/crash-reporter.ts:104]: this could be simplified to
```
this.appState = {...this.readStateFile, code};
if (this.appState.code !== EAppState.Starting) {
      this.appState.detected = '';
}
```

Comment (summeroff) [reply app/services/crash-reporter.ts:103]: It does not let me to use ... in this way. 

Removed unnecessary preinitialization.
